### PR TITLE
Add react-focus-lock and react-remove-scroll

### PR DESCRIPTION
**Pull request by theKashey**
React-Focus-Lock fixes:
- #83, it accepts _focus_ events via React Event System, thus count portaled content as own content.
- unreported issue about nested dialogs. Once second dialog is closed lock is no more active.
- `initialFocusRef` was not drilled to Dialog

Scroll-Lock(react-remove-scroll) fixes:
- #98, `position:fixed` elements have to disable body scroll on iOS
- unreported issue, alerts and modals should lock scroll(by design).
- also support portals, nested locks and so on

+ a bit simpler code about focus management
+ a few examples to prove/test new stuff

### CODE REVIEW

File: packages/dialog/examples/long-content.example.js [hunk 1]
```diff
@@ -13,6 +13,16 @@ export let Example = () => (
           Show Dialog
         </button>
 
+        {new Array(20)
+          .fill(1)
+          .map((x, index) => (
+            <div
+              style={{ position: "absolute", top: (index + 1) * 100 + "px" }}
+            >
+              - scroll -
```
Comment (theKashey) [inline packages/dialog/examples/long-content.example.js]: Just to test scroll lock.

**Comment by ryanflorence** [comment]
merging and releasing this today, thanks for your patience and for the code :D

**Comment by ryanflorence** [comment]
done, @reach/dialog@0.2.0

Thanks again!

**Comment by theKashey** [comment]
Happy to reach reach. From now a part of your issues are mine :)

**Comment by karlhorky** [comment]
I guess this causes the issue that I'm experiencing with the scroll locking not working: reach/reach-ui#161 (upstream theKashey/react-remove-scroll#4).

**Comment by mkarajohn** [comment]
>unreported issue about nested dialogs. Once second dialog is closed lock is no more active. 

Was it [this](https://github.com/reach/reach-ui/issues/130) one? If that's the case I would have loved it if the issue had been mentioned here so I would have been notified that this was fixed.

**Comment by theKashey** [comment]
@mkarajohn - yeah, that was it.

**Comment by mkarajohn** [comment]
Huh, that's a bummer. Why was it considered an "unreported issue"?

**Comment by theKashey** [comment]
Cos you added it __months__ after I've created this PR? And a few weeks before it got merged :)

**Comment by mkarajohn** [comment]
Hah, holy shit, you are right. For some reason the merge date registered as the opening date in my brain, no kidding. Way to make an ass out of myself 👏

**Pull request merged**